packages/labs/context/src/test/content-provider_test.ts

```diff
@@ -134,3 +136,125 @@ suite('context-provider', () => {
+    // to help re-use same context object, yet target a destination
+    if (this.userId === userId) {
+      const { firstName = '', lastName = '', age = 0 } = rest ?? {}
+      this.firstName = firstName;
+      this.lastName = lastName;
+      this.age = age;
+    }
+  }
+
+  protected render(): TemplateResult {
+    const isNoLongerEmpty = ['firstName', 'lastName'].filter(k  => Reflect.get(this, k) !== '').length >= 2
+    return html`Hi, ${cache(isNoLongerEmpty ?
+      html`my name is <span id="firstName">${this.firstName}</span> <span id="lastName">${this.lastName}</span>, and I am <span id="age">${this.age}</span>` :
+      html`sir`)}`;
+  }
+}
+
+customElements.define('user-info-badge-context-consumer', UserInfoBadgeContextConsumer);
+customElements.define('user-info-context-provider', UserInfoContextProvider);
+
+suite('context-provider when many consumer different identities', () => {
+  let provider: UserInfoContextProvider;
+  let consumers: NodeListOf<UserInfoBadgeContextConsumer>;
+
+  const aliceUserId = 'alice'
+  const aliceUserInfo: IUserInfoContext = Object.freeze({ userId: aliceUserId, firstName: 'Alice', lastName: 'Carroll', age: 41 })
+
+  const bobUserId = 'bob'
+  const bobUserInfo: IUserInfoContext = Object.freeze({ userId: bobUserId, firstName: 'Bob', lastName: 'Dylan', age: 40 })
+
+  setup(async () => {
+    const container = document.createElement('div');
+    container.innerHTML = `
+      <user-info-context-provider>
+        <user-info-badge-context-consumer data-user-id="alice"></user-info-badge-context-consumer>
+        <!-- many more ... -->
+        <user-info-badge-context-consumer data-user-id="bob"></user-info-badge-context-consumer>
+      </user-info-context-provider>
+    `;
+    document.body.appendChild(container);
+
+    provider = container.querySelector(
+      'user-info-context-provider'
+    ) as UserInfoContextProvider;
+    consumers = provider.querySelectorAll(
+      'user-info-badge-context-consumer'
+    ) as NodeListOf<UserInfoBadgeContextConsumer>;
+    assert.isDefined(consumers);
+    assert.lengthOf(consumers, 2);
+    assert.isDefined(provider);
+  });
+
+  test(`alpha`, async () => {
+    assert.strictEqual(consumers[0].firstName, '')
+    assert.strictEqual(consumers[1].firstName, '')
+    await consumers[1].updateComplete
+    assert.equal(consumers[1].shadowRoot!.textContent, 'Hi, sir');
+  })
+
+  test(`bravo`, async () => {
+    // Database data pulling, things happening, aggregating data.
+    provider.setValue({ ...aliceUserInfo })
+    provider.setValue({ ...bobUserInfo })
+    const [alice, bob] = consumers
+    assert.strictEqual(alice.firstName, 'Alice')
+    assert.strictEqual(alice.lastName, 'Carroll')
+    assert.strictEqual(bob.firstName, 'Bob')
+    assert.strictEqual(bob.lastName, 'Dylan')
+    await alice.updateComplete
+    assert.equal(alice.shadowRoot!.textContent, 'Hi, my name is Alice Carroll, and I am 41');
+  })
```

Comment:
Another use-case for this pattern would be when we might want to use for CSS configurability for passing design tokens around.

For example, that's what [React’s **RecoilJS** core-concepts](https://recoiljs.org/docs/introduction/core-concepts/) gives as example for setting a fontSize.

```ts
// Modified from https://recoiljs.org/docs/introduction/core-concepts
export const fontSizeState = atom({  key: 'fontSizeState',  default: 14, discriminator: '' });
```

In ContextAPI, we could do like this:

```ts
// module: @app/contexts
export interface IFontSize {
  discriminator: string
  size: string
}

export const FontSizeContext = createContext<IFontSize>('fontSizeState', Object.freeze({ size: 14,               discriminator: '' }));
```

Then, an hypothetical consumer:

```ts
import { FontSizeContext } from '@app/contexts'
import type { IFontSize } from '@app/contexts'

// This was yanked from a conversation, untested.
export SomeComponent extends LitElement {
  static properties = {
    fontSizeForTitle: { type: String, attribute: false },
    fontSizeForSubMenu: { type: String, attribute: false },
  }
  static get styles = css`
    h1 {
      font-size: ${this.fontSizeForTitle};
    }
    nav.submenu {
      font-size: ${this.fontSizeForSubMenu};
    }
  `

  private fontSizeForTitle = 0
  private fontSizeForSubMenu = 0

  connectedCallback() {
    this.dispatchEvent(
      new ContextEvent(
        FontSizeContext,
        callback: (data: IFontSize) => {
          const { fontSize = 0, discriminant } = data;
          let changingFontSize: number
          switch (discriminant) {
            case 'Title':
              changingFontSize = fontSize
              break;
            case 'SubMenu':
              changingFontSize = fontSize
              break;
          }
          if (changingFontSize) {
            this.fontSize = fontSize;
          }
        }
      )
    );
  }
}
```